executor/src/witgen/affine_expression.rs

```diff
@@ -93,14 +94,14 @@ impl AffineExpression {
                     } else {
                         -self.offset / *c
                     }),
-                )])
+                )]))
             }
             (Some(_), Some(_)) => Err("Too many variables in linear constraint."
                 .to_string()
                 .into()),
             (None, None) => {
                 if self.offset == 0.into() {
-                    Ok(vec![])
+                    Ok(EvalValue::incomplete(vec![]))
                 } else {
                     Err(ConstraintUnsatisfiable(String::new()))
```

Comment:
Maybe we should also re-think this. ConstraintUnsatisfiable is the only real error, the others are just "cannot progress, but it might still be fine".

Reply:
So maybe the other error should be "incomplete with explanation".

Reply:
That makes sense. I suspect we should get the structure right in this PR and add more data to the actual `IncompleteCause` in another PR.

Reply:
Ended up implementing a lot of this here, but not actually passing that information back. Should be easy to add now.